coaster/sqlalchemy.py

```diff
@@ -166,6 +166,26 @@ def __init__(self, *args, **kw):
         if not self.name:
             self.make_name()
 
+    @classmethod
+    def get(cls, name):
+        """Get an instance matching the name"""
+        return cls.query.filter_by(name=name).one_or_none()
+
+    @classmethod
+    def upsert(cls, name, **fields):
+        """Insert or update an instance"""
+        instance = cls.get(name)
+        if instance:
+            for f in fields:
+                if hasattr(instance, f):
+                    setattr(instance, f, fields[f])
+                else:
+                    raise TypeError("'{arg}' is an invalid keyword argument for {instance_type}".format(arg=f, instance_type=instance.__class__.__name__))
```

Comment:
The behavior of setting the fields of an instance, in L179 - L183 is repeated in `BaseNameMixin` and `BaseScopedNameMixin`. Does it make sense to move this behavior into a method, say `set_fields`, in `BaseMixin`? Apart for DRYing up `BaseNameMixin` and `BaseScopedNameMixin`, this would also let a class that inherits directly from `BaseMixin`, reuse this behavior.